aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -253,7 +253,8 @@ Tensor polar(const Tensor& abs, const Tensor& angle) {
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ empty ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Tensor empty_cpu(IntArrayRef size, c10::optional<ScalarType> dtype_opt, c10::optional<Layout> layout_opt,
                  c10::optional<Device> device_opt, c10::optional<bool> pin_memory_opt, c10::optional<c10::MemoryFormat> memory_format_opt) {
-  return at::detail::empty_cpu(size, dtype_opt, layout_opt, device_opt, pin_memory_opt, memory_format_opt);
+  Tensor result = at::detail::empty_cpu(size, dtype_opt, layout_opt, device_opt, pin_memory_opt, memory_format_opt);
+  return maybe_fill_empty_deterministic(result);
```

Comment:
Given how much this function is in the hot path, we probably want to pull back the `at::globalContext().deterministicAlgorithms()` check to here, to avoid the indirection (and ref bump) that may be caused by the call to the function if it can't be inlined.